modules/cardano-node.nix

```diff
@@ -167,6 +158,28 @@ in {
       allowedUDPPortRanges = [ { from = 1024; to = 65000; } ];
     };
 
+    # Workaround for CSL-1029
+    # Reboot cardano-node every day, offset by node id (in 4 minute intervals)
+    systemd.services.cardano-restart = let
+      getDailyTime = testIndex: let
+          # how many minutes between each node restarting
+          interval = 60;
+          minute = mod (testIndex * interval) 60;
+          hour = mod ((testIndex * interval) / 60) 24;
+        in "${toString hour}:${toString minute}";
+      getHourlyTime = testIndex: let
+          # how many minutes between each node restarting
+          interval = 4;
+          minute = mod (testIndex * interval) 60;
+        in "*:${toString minute}";
+      getInterval = testIndex: if testIndex < 3 then (getDailyTime testIndex) else (getHourlyTime testIndex);
+    in {
+      script = ''
+        /run/current-system/sw/bin/systemctl restart cardano-node
+      '';
+      startAt = getDailyTime cfg.testIndex;
+    };
+
```

Comment:
Ah, I see. Thanks for clarifying.

It's compatible with https://github.com/input-output-hk/iohk-nixops/blob/staging/modules/cardano-node.nix#L170-L194, right?